examples/js/renderers/SoftwareRenderer.js

```diff
@@ -704,68 +716,65 @@ THREE.SoftwareRenderer = function ( parameters ) {
 			// 2      3
-			var	mpV12 = new THREE.Vector4(),
-				mpV23 = new THREE.Vector4(),
-				mpV31 = new THREE.Vector4(),
-				mpN12 = new THREE.Vector3(),
-				mpN23 = new THREE.Vector3(),
-				mpN31 = new THREE.Vector3(),
-				mpUV12 = new THREE.Vector2(),
-				mpUV23 = new THREE.Vector2(),
-				mpUV31 = new THREE.Vector2(),
-				tempFace = { vertexNormalsModel : []
-							, color : { r: face.color.r, g: face.color.b, b: face.color.b } };
-
-			var weight;
-
-			weight = (1 + v2.z) * (v2.w / v1.w) / (1 + v1.z);
-			mpUV12.copy( uv1 ).multiplyScalar( weight ).add( uv2 ).multiplyScalar( 1 / (weight + 1) );
-			weight = (1 + v3.z) * (v3.w / v2.w) / (1 + v2.z);
-			mpUV23.copy( uv2 ).multiplyScalar( weight ).add( uv3 ).multiplyScalar( 1 / (weight + 1) );
-			weight = (1 + v1.z) * (v1.w / v3.w) / (1 + v3.z);
-			mpUV31.copy( uv3 ).multiplyScalar( weight ).add( uv1 ).multiplyScalar( 1 / (weight + 1) );
-
-			mpV12.copy( v1 ).add( v2 ).multiplyScalar( 0.5 );
-			mpV23.copy( v2 ).add( v3 ).multiplyScalar( 0.5 );
-			mpV31.copy( v3 ).add( v1 ).multiplyScalar( 0.5 );
+			var tempFace = { vertexNormalsModel : [], 
+						color : face.color };
+
+			if ( bHasUV ) {
+				var weight;
+
+				weight = (1 + v2.z) * (v2.w / v1.w) / (1 + v1.z);
+				texCoord1.copy( uv1 ).multiplyScalar( weight ).add( uv2 ).multiplyScalar( 1 / (weight + 1) );
+				weight = (1 + v3.z) * (v3.w / v2.w) / (1 + v2.z);
+				texCoord2.copy( uv2 ).multiplyScalar( weight ).add( uv3 ).multiplyScalar( 1 / (weight + 1) );
+				weight = (1 + v1.z) * (v1.w / v3.w) / (1 + v3.z);
+				texCoord3.copy( uv3 ).multiplyScalar( weight ).add( uv1 ).multiplyScalar( 1 / (weight + 1) );
+			
+				uv1 = texCoord1;
+				uv2 = texCoord2;
+				uv3 = texCoord3;
+			}
+			
+			vector1.copy( v1 ).add( v2 ).multiplyScalar( 0.5 );
+			vector2.copy( v2 ).add( v3 ).multiplyScalar( 0.5 );
+			vector3.copy( v3 ).add( v1 ).multiplyScalar( 0.5 );
 
 			if( bHasNormal ) {
-				mpN12.copy( face.vertexNormalsModel[ 0 ] ).add( face.vertexNormalsModel[ 1 ] ).normalize();
-				mpN23.copy( face.vertexNormalsModel[ 1 ] ).add( face.vertexNormalsModel[ 2 ] ).normalize();
-				mpN31.copy( face.vertexNormalsModel[ 2 ] ).add( face.vertexNormalsModel[ 0 ] ).normalize();
+				normal1.copy( face.vertexNormalsModel[ 0 ] ).add( face.vertexNormalsModel[ 1 ] ).normalize();
+				normal2.copy( face.vertexNormalsModel[ 1 ] ).add( face.vertexNormalsModel[ 2 ] ).normalize();
+				normal3.copy( face.vertexNormalsModel[ 2 ] ).add( face.vertexNormalsModel[ 0 ] ).normalize();
 			}
 
 			// a
 			if( bHasNormal ) {
 				tempFace.vertexNormalsModel[ 0 ] = face.vertexNormalsModel[ 0 ];
-				tempFace.vertexNormalsModel[ 1 ] = mpN12;
-				tempFace.vertexNormalsModel[ 2 ] = mpN31;
+				tempFace.vertexNormalsModel[ 1 ] = normal1;
+				tempFace.vertexNormalsModel[ 2 ] = normal3;
 			}
-			drawTriangle( v1, mpV12, mpV31, uv1, mpUV12, mpUV31, shader, tempFace, material );
+			drawTriangle( v1, vector1, vector3, uv1, uv1, uv3, shader, tempFace, material );
```

Comment:
there is something wrong here, you pass uv1 twice. same goes for the calls below.